### PR TITLE
Add support for India Residency

### PR DESCRIPTION
**Pull request by myronkaifung**
- Changed `eu` param to `residency`
- `Residency` is a str that accepts 'us', 'eu', or 'in'. Default value = "us"
- Updated readme to reflect the above

### CODE REVIEW

**Review by austinpray-mixpanel — APPROVED** [review]

**Review by austinpray-mixpanel — CHANGES_REQUESTED** [review]
Future proofing discussion happening in slack. Defer to @dongjae93

File: src/mixpanel_utils/__init__.py [hunk 1]
```diff
@@ -106,17 +106,23 @@ def __init__(
...
+        self.residency = residency
+        residency_values = ["us","eu","in"]
+        assert self.residency in residency_values, "residency value must be 'us', 'eu, or 'in'!"
         self.raw_api = (
-            "https://data.mixpanel.com/api"
-            if eu is False
-            else "https://data-eu.mixpanel.com/api"
+            "https://data.mixpanel.com/api" if residency == "us"
+            else "https://data-eu.mixpanel.com/api" if residency == "eu"
+            else "https://data-in.mixpanel.com/api"
```
Comment (dongjae93) [inline src/mixpanel_utils/__init__.py:115]: let's follow what Ted suggested and make this future proof. The validation logic ensures we will have valid residency value so we can do something like 
if us:  "https://data.mixpanel.com/api"
else: f"https://data-{residency}.mixpanel.com/api"

File: src/mixpanel_utils/__init__.py [hunk 2]
```diff
@@ -106,17 +106,23 @@ def __init__(
...
         )
         self.import_api = (
-            "https://api.mixpanel.com" if eu is False else "https://api-eu.mixpanel.com"
+            "https://api.mixpanel.com" if residency == "us" 
+            else "https://api-eu.mixpanel.com" if residency == "eu"
+            else "https://api-in.mixpanel.com"
```
Comment (dongjae93) [inline src/mixpanel_utils/__init__.py:120]: same as above

File: src/mixpanel_utils/__init__.py [hunk 3]
```diff
@@ -106,17 +106,23 @@ def __init__(
...
         )
         self.formatted_api = (
-            "https://mixpanel.com/api" if eu is False else "https://eu.mixpanel.com/api"
+            "https://mixpanel.com/api" if residency == "us" 
+            else "https://eu.mixpanel.com/api" if residency == "eu"
+            else "https://in.mixpanel.com/api"
```
Comment (dongjae93) [inline src/mixpanel_utils/__init__.py:125]: same as above

**Review by dongjae93 — CHANGES_REQUESTED** [review]

**Comment by myronkaifung** [comment]
@dongjae93 would you please review again?

**Review by dongjae93 — APPROVED** [review]
LGTM!

**Comment by myronkaifung** [comment]
Can you merge for me? I have no write access

**Pull request merged**